src/client/executor.rs

```diff
@@ -82,16 +82,7 @@ impl Client {
     ) -> Result<T::O> {
         let server = self.select_server(op.selection_criteria()).await?;
 
-        let mut conn = match server.checkout_connection().await {
-            Ok(conn) => conn,
-            Err(err) => {
-                self.inner
-                    .topology
-                    .handle_pre_handshake_error(err.clone(), server.address.clone())
-                    .await;
-                return Err(err);
-            }
-        };
+        let mut conn = server.pool.check_out().await?;
```

Comment:
In order to unify establishment error handling, I removed any special handling from in the executor and had the monitor do it instead by listening to a channel broadcasted by the pool itself. This allows us to use the same error handling regardless of whether a connection was established as part of operation execution or via the background thread.